### ISSUE TITLE
Trying to use Netlib (Spark) with OpenBLAS

### ISSUE THREAD

**Issue by RoiViber** [fommil/netlib-java#88]
Hi. 
I'm trying for some time to configure Spark to run with OpenBLAS libraries I build on AWS hardware. It seems that I can manage to work the BLAS part successfully only when setting:
--driver-java-options  -Dcom.github.fommil.netlib.NativeSystemBLAS.natives=< path to: libopenblas.so >

But what about LAPACK? I've read (and the make procedure also tells you) that by building OpenBLAS it  also assembles its version of LAPACK in (possibly in libopenblas.a?)

So I tried both:
-Dcom.github.fommil.netlib.NativeSystemLAPACK.natives= < path to: libopenblas.so >
-Dcom.github.fommil.netlib.NativeSystemLAPACK.natives= < path to: libopenblas.a >

The first is throwing an error regarding not finding some lapack function (which tells me, as far as I understand that libopenblas.so does not include lapack), but the second one throws some different error that tells me: "You have loaded ... library which might have disabled stack guard", and also something about EXECSTACKing -C <library>
when I try:
execstack -C <libopenblas.a>
it tells me it isn't an ETF library.

Any idea where is OpenBLAS's lapack library??

**Comment by fommil** [comment]
this is all documented in https://github.com/fommil/netlib-java#machine-optimised-system-libraries

If you are passing your `libopenblas.so` as a parameter to `com.github.fommil.netlib.NativeSystemBLAS.natives` you have not fully grokked the documentation, as that will not work.

**Comment by RoiViber** [comment]
Hi.

I read this one, and have actually managed to make this work also by linking (using alternatives) and using the JNI loader. My problem still remains. 
I know that i should link <libblas.so.3> to <libopenblas.so> 
but what about <liblapack.so.3> ?

the OpenBLAS build doesn't provide me with anything that have "lapack" in its name. The only files I get from MAKE-INSTALLing the OpenBLAS are:
libopenblas.so
libopenblas.a
libopenblas*sandybridge*.so
libopenblas*sandybridge*.a

**Comment by fommil** [comment]
ask your GNU/Linux distribution or sysadmin to provide you with the ability to create these links

**Comment by fommil** [comment]
probably creating a link from `liblapack.so.3` to `libopenblas.so` will work, but you don't pass these to the JVM like you're doing: that's the `jni` library that goes there.

**Comment by RoiViber** [comment]
OK. I understand that the "jni" library does the linking through a temporary .so file it creates for the Spark run (in /tmp). I also got it linking right - the < temp netlib's .so file> contains the same links as the <libopenblas.so> file (i ldd both of them), except that i got the following error when trying matrix multiplication in spark-shell:
/usr/java/latest/bin/java: symbol lookup error: /tmp/jniloader6728254881598867695netlib-native_system-linux-x86_64.so: undefined symbol: cblas_dgemm

And now I really believe I did everything written in the manual.

I finally got it solved by adding:
export LD_PRELOAD=< libopenblas.so >

Thanks for everything, especially the quick reply. I really appreciate that!
